web/src/beta/lib/core/StoryPanel/Block/builtin/Camera/index.tsx

```diff
@@ -0,0 +1,118 @@
+
+import { getFieldValue } from "../../../utils";
+import { CommonProps as BlockProps } from "../../types";
+import usePropertyValueUpdate from "../common/usePropertyValueUpdate";
+import BlockWrapper from "../common/Wrapper";
+
+import CameraEditor, { Props as EditorProps } from "./Editor";
+
+export type Props = BlockProps;
+
+const CameraBlock: React.FC<Props> = ({ block, isSelected, ...props }) => {
+  const {
+    handlePropertyValueUpdate,
+    handleAddPropertyItem,
+    handleRemovePropertyItem,
+    handleMovePropertyItem,
+  } = usePropertyValueUpdate();
+
+  const visualizer = useVisualizer();
+
+  const currentCamera = useReactiveVar(currentCameraVar);
+  const handleFlyTo = useMemo(() => visualizer.current?.engine.flyTo, [visualizer]);
+
+  const items = useMemo(
+    () => getFieldValue(block?.property?.items ?? [], "") as EditorProps["items"],
+    [block?.property?.items],
+  );
+
+  const handleAddItem = useCallback(() => {
+    const schemaGroup = block?.property?.items?.find(i => i.schemaGroup === "default")?.schemaGroup;
+    if (!block?.property?.id || !schemaGroup) return;
+    handleAddPropertyItem(block.property.id, schemaGroup);
+  }, [block?.property?.id, block?.property?.items, handleAddPropertyItem]);
+
+  const handleUpdate = useCallback(
+    (
+      itemId: string,
+      fieldId: string,
+      fieldType: keyof ValueTypes,
+      updatedValue: ValueTypes[keyof ValueTypes],
+    ) => {
+      const schemaGroup = block?.property?.items?.find(
+        i => i.schemaGroup === "default",
+      )?.schemaGroup;
+      if (!block?.property?.id || !itemId || !schemaGroup) return;
+
+      handlePropertyValueUpdate(
+        schemaGroup,
+        block?.property?.id,
+        fieldId,
+        fieldType,
+        itemId,
+      )(updatedValue);
+    },
+    [block?.property?.id, block?.property?.items, handlePropertyValueUpdate],
+  );
+
+  const handleRemoveItem = useCallback(
+    (itemId: string) => {
+      const schemaGroup = block?.property?.items?.find(
+        i => i.schemaGroup === "default",
+      )?.schemaGroup;
+      if (!block?.property?.id || !itemId || !schemaGroup) return;
+
+      handleRemovePropertyItem(block.property.id, schemaGroup, itemId);
+    },
+    [block?.property?.id, block?.property?.items, handleRemovePropertyItem],
+  );
+
+  const handleMoveItem = useCallback(
+    ({ id }: { id: string }, index: number) => {
+      const schemaGroup = block?.property?.items?.find(
+        i => i.schemaGroup === "default",
+      )?.schemaGroup;
+      if (!block?.property?.id || !id || !schemaGroup) return;
+
+      handleMovePropertyItem(block.property.id, schemaGroup, { id }, index);
+    },
+    [block?.property?.id, block?.property?.items, handleMovePropertyItem],
+  );
+
+  // if there's no item add 1 button.
+  useEffect(() => {
+    if (items.length === 0) {
+      handleAddItem();
+      return;
+    }
+  }, [items.length, handleAddItem]);
```

Comment:
I feel like we should add this to block creation in the storytellingApi, checking if its camera block (and in the future, layer switch block, etc) and if so, creating an item.

Reply:
I'll do it when refactoring and taking the commmon parts of button away. Have put a todo!